src/features/PowerShellNotebooks.ts

```diff
@@ -0,0 +1,220 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { CommentType } from '../settings';
+import { IFeature, LanguageClient } from '../feature';
+import { EvaluateRequestType } from './Console';
+import Settings = require("../settings");
+
+export class PowerShellNotebooksFeature implements vscode.NotebookContentProvider, vscode.NotebookKernel, IFeature {
+
+    private readonly showNotebookModeCommand: vscode.Disposable;
+    private readonly hideNotebookModeCommand: vscode.Disposable;
+    private languageClient: LanguageClient;
+
+    private _onDidChangeNotebook = new vscode.EventEmitter<vscode.NotebookDocumentEditEvent>();
+    onDidChangeNotebook: vscode.Event<vscode.NotebookDocumentEditEvent> = this._onDidChangeNotebook.event;
+    kernel?: vscode.NotebookKernel;
+
+    constructor() {
+        this.kernel = this;
```

Comment:
Yes the vscode API uses it under the hood. I could potentially have 2 classes, one that implements `vscode.NotebookContentProvider` and one that implements `vscode.NotebookKernel`... but I felt one was fine, since `vscode.NotebookKernel` isn't that big of an interface.

Reply:
Ah gotcha. This makes sense -- one of the great advantages of interfaces

Reply:
Also, this originally was one interface, but they broke it up into 2 recently.